File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
#1260

def dfs(c):
    v[c]=1
    ans_dfs.append(c)
    for n in adj[c]:
        if v[n]==0: 
            dfs(n) #방문 안한 노드면 재귀함수 호출
            
def bfs(s):
    q=[]
    q.append(s) #큐에 초기 데이터 삽입
    
    v[s]=1
    ans_bfs.append(s)
    
    while q :
        c=q.pop(0) #큐에서 데이터 하나 빼주기
        for n in adj[c]:
            if v[n]==0: #방문 안한 노드라면 큐에 삽입
                q.append(n)
                ans_bfs.append(n)
                v[n]=1


            
N, M, V = map(int, input().split())
adj= [[] for i in range(N+1)]
for i in range(M) :
    s, e = map(int, input().split())
    adj[s].append(e)
    adj[e].append(s)

for i in range(1, N+1):
    adj[i].sort()

v=[0]*(N+1)
ans_dfs=[]
dfs(V)

v=[0]*(N+1)
ans_bfs=[]
bfs(V)

print(*ans_dfs)    #unpacking
print(*ans_bfs)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
	static List<List<Integer>> adj;
	static boolean[] v;
	
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		int s = Integer.parseInt(st.nextToken());
		
		adj = new ArrayList<>();
		for (int i = 0; i <= n; i++) {
			adj.add(new ArrayList<>());
		}
		
		for (int i = 0; i < m; i++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			
			adj.get(a).add(b);
			adj.get(b).add(a);
		}
		
		for (int i = 1; i <= n; i++) {
			Collections.sort(adj.get(i));
		}
		
		v = new boolean[n + 1];
		dfs(s);
		System.out.println();
		Arrays.fill(v, false);
		bfs(s);
		
	}

	static void dfs(int s) {
		v[s] = true;
		System.out.print(s + " ");
		
		for (int nxt : adj.get(s)) {
			if (!v[nxt]) {
				dfs(nxt);
			}
		}
	}

	static void bfs(int s) {
		Queue<Integer> q = new ArrayDeque<>();
		v[s] = true;
		q.add(s);
		
		while(!q.isEmpty()) {
			int c = q.poll();
			System.out.print(c + " ");

			for (int nxt : adj.get(c)) {
				if (!v[nxt]) {
					v[nxt] = true;
					q.add(nxt);
				}
			}
		}
	}
}
